### PR TITLE
Fix file upload were gzip is larger

### PR DESCRIPTION
**Pull request by hishnash**
we need to ensure that we don't compress arrays if the binary size when compressed is larger than the size before compression.

### CODE REVIEW

File: lfview/client/session.py [hunk 1]
```diff
@@ -432,13 +432,22 @@ def _upload(
         Use :code:`upload`, :code:`upload_slide`, or :code:`upload_feedback`
         instead.
         """
+        is_compressed = False
```
Comment (morepork) [inline lfview/client/session.py]: get rid of this

File: lfview/client/session.py [hunk 2]
```diff
@@ -432,13 +432,22 @@ def _upload(
...
+            raw_array = resource.array.tobytes()
+            compressed_arr = _compress_bytes(raw_array)
+            data_to_upload = None
+            # only set compressed data if the data is smaller
+            if len(compressed_arr) < len(raw_array):
+                json_dict['content_length'] = len(compressed_arr)
+                json_dict['content_encoding'] = 'gzip'
+                data_to_upload = compressed_arr
+            else:
+                json_dict['content_length'] = len(raw_array)
```
Comment (morepork) [inline lfview/client/session.py:449]: add a note about lack of content ecndoing

**Review by morepork — APPROVED** [review]

**Review by fwkoch — APPROVED** [review]
This looks great. Is it safe to say `compressed_size > raw_size` is a pretty limited edge case? But certainly worth supporting.

**Comment by hishnash** [comment]
> Is it safe to say compressed_size > raw_size is a pretty limited edge case? But certainly worth supporting.

yes this really will only happened for very small files or very very un-compressable data.  

the reason we need to do this is our server will reject these requests if the compressed size is larger than the un-compressed size.

**Review by jrgirvan — APPROVED** [review]

**Pull request merged**